packages/core/admin/admin/src/hooks/useSettingsForm/tests/index.test.js

```diff
@@ -0,0 +1,125 @@
+import * as React from 'react';
+import { setupServer } from 'msw/node';
+import { rest } from 'msw';
+import { renderHook, act } from '@testing-library/react-hooks';
+import { IntlProvider } from 'react-intl';
+import omit from 'lodash/omit';
+import useSettingsForm from '../index';
+
+const toggleNotification = jest.fn();
+
+jest.mock('@strapi/helper-plugin', () => ({
+  ...jest.requireActual('@strapi/helper-plugin'),
+  useNotification: jest.fn().mockImplementation(() => toggleNotification),
+  useOverlayBlocker: () => ({ lockApp: jest.fn(), unlockApp: jest.fn() }),
+}));
+
+jest.mock('../../../utils', () => ({
+  ...jest.requireActual('../../../utils'),
+  checkFormValidity: () => (null),
+}));
+
+jest.mock('lodash/omit');
+
+const handlers = [
+  rest.put('*/providers/options', (req, res, ctx) =>
+    res(
+      ctx.status(200),
+      ctx.json({
+        data: {
+          autoRegister: false,
+          defaultRole: "1",
+          ssoLockedRoles: ["1","2","3"]
+        }
+      })
+    )
+  ),
+  rest.get('*/providers/options', (req, res, ctx) =>
+    res(
+      ctx.status(200),
+      ctx.json({
+        data: {
+          autoRegister: false,
+          defaultRole: "1",
+          ssoLockedRoles: ["1","2"]
+        }
+      })
+    )
+  )
+];
+
+const server = setupServer(...handlers);
+
+const setup = (...args) =>
+  renderHook(() => useSettingsForm(...args), {
+    wrapper({ children }) {
+      return (
+        <IntlProvider locale="en" messages={{}}>
```

Comment:
Why do you have to wrap in `IntlProvider`? Isn't the `useNotification` mock sufficient?

Reply:
done